src/frontend/src/common/components/library/data_subview/index.tsx

```diff
@@ -102,6 +129,115 @@ const DataSubview: FunctionComponent<Props> = ({
     searching: false,
   });
 
+  const [checkedSamples, setCheckedSamples] = useState<any[]>([]);
+  const [isHeaderChecked, setIsHeaderChecked] = useState<boolean>(false);
+  const [showCheckboxes, setShowCheckboxes] = useState<boolean>(false);
+  const [isHeaderIndeterminant, setHeaderIndeterminant] =
+    useState<boolean>(false);
+  const [open, setOpen] = useState(false);
+  const [isDownloadDisabled, setDownloadDisabled] = useState<boolean>(true);
+  const [failedSamples, setFailedSamples] = useState<string[]>([]);
+  const [downloadFailed, setDownloadFailed] = useState<boolean>(false);
+  const [isMetadataSelected, setMetadataSelected] = useState<boolean>(false);
+  const [isFastaSelected, setFastaSelected] = useState<boolean>(false);
+  const [isFastaDisabled, setFastaDisabled] = useState<boolean>(false);
+
+  const handleDownloadClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleDownloadClose = () => {
+    setOpen(false);
+    setMetadataSelected(false);
+    setFastaSelected(false);
+  };
+
+  useEffect(() => {
+    // add all samples if header checkbox is selected
+    if (isHeaderChecked) {
+      const allPublicIds: any[] = [];
+      const failedSamples: any[] = [];
+      for (const key in data) {
+        allPublicIds.push(data[key as any].publicId);
+        if (data[key as any].CZBFailedGenomeRecovery) {
+          failedSamples.push(data[key as any].publicId);
+        }
+      }
+      setCheckedSamples(allPublicIds);
+      setFailedSamples(failedSamples);
+    } else {
+      setFailedSamples([]);
+      setCheckedSamples([]);
+    }
+  }, [isHeaderChecked, data]);
+
+  useEffect(() => {
+    // Only show checkboxes on the sample datatable
+    if (viewName === "Samples") {
+      setShowCheckboxes(true);
+    }
+  }, [viewName]);
+
+  useEffect(() => {
+    // determine if mixed state (user has custom selected samples)
+    if (data) {
+      const sizeData: number = Object.keys(data).length;
```

Comment:
nit: we don't need to type `: number` here, since TS is smart enough to infer from calling property getter `.length`!